frontends/web/src/routes/exchange/bitrefill.tsx

```diff
@@ -0,0 +1,235 @@
+
+  useEffect(() => {
+    if (config) {
+      setAgreedTerms(config.frontend.skipBitrefillWidgetDisclaimer);
+    }
+  }, [config]);
+
+  const onResize = useCallback(() => {
+    if (resizeTimerID.current) {
+      clearTimeout(resizeTimerID.current);
+    }
+    resizeTimerID.current = setTimeout(() => {
+      if (containerRef.current) {
+        setHeight(containerRef.current.offsetHeight);
+      }
+    }, 200);
+  }, []);
+
+  useEffect(() => {
+    onResize();
+    window.addEventListener('resize', onResize);
+    return () => {
+      window.removeEventListener('resize', onResize);
+      if (resizeTimerID.current) {
+        clearTimeout(resizeTimerID.current);
+      }
+    };
+  }, [onResize]);
+
+  const handleMessage = useCallback(async (event: MessageEvent) => {
+    if (
+      !account
+      || !bitrefillInfo?.success
+      || ![getURLOrigin(bitrefillInfo.url), 'https://embed.bitrefill.com'].includes(event.origin)
+    ) {
+      return;
+    }
+
+    const data = typeof event.data === 'string' ? JSON.parse(event.data) : event.data;
+
+    switch (data.event) {
+    case 'request-configuration': {
+      event.source?.postMessage({
+        event: 'configuration',
+        ref: bitrefillInfo.ref,
+        utm_source: 'BITBOX',
+        theme: isDarkMode ? 'dark' : 'light',
+        hl: i18n.resolvedLanguage ? localeMapping[i18n.resolvedLanguage] : 'en',
+        paymentMethods: account.coinCode ? coinMapping[account.coinCode] : 'bitcoin',
+        refundAddress: bitrefillInfo.address,
+        // Option to keep pending payment information longer in session, defaults to 'false'
+        paymentPending: 'true',
+        // Option to show payment information in the widget, defaults to 'true'
+        showPaymentInfo: 'true'
+      }, {
+        targetOrigin: event.origin
+      });
+      break;
+    }
+    case 'payment_intent': {
+      // User clicked "Pay" in checkout
+      const {
+        invoiceId,
+        paymentMethod,
+        paymentAmount,
+        paymentAddress,
+      } = data;
+
+      const parsedAmount = await parseExternalBtcAmount(paymentAmount.toString());
+      if (!parsedAmount.success) {
+        alertUser(t('unknownError', { errorMessage: 'Invalid amount' }));
+        return;
+      }
+      // Ensure expected payment method matches account
+      if (coinMapping[account.coinCode] !== paymentMethod) {
+        alertUser(t('unknownError', { errorMessage: 'Payment method mismatch' }));
+      }
+
+      const txInput: TTxInput = {
+        address: paymentAddress,
+        amount: parsedAmount.amount,
+        // Always use highest fee rate for Bitrefill spend
+        useHighestFee: true,
+        sendAll: 'no',
+        selectedUTXOs: [],
+        paymentRequest: null
+      };
+
+      let result = await proposeTx(code, txInput);
+      if (result.success) {
+        const txNote = t('generic.paymentRequestNote', {
+          name: 'Bitrefill',
+          invoiceId,
```

Comment:
oups in the other PR the placeholder is called orderId

Reply:
Updated